contrib/cscope/packages.el

```diff
@@ -0,0 +1,61 @@
+;;; packages.el --- cscope Layer packages File for Spacemacs
+;;
+;; Copyright (c) 2012-2014 Sylvain Benner
+;; Copyright (c) 2014-2015 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(setq cscope-packages '(evil-jumper
+                        helm-cscope
+                        xcscope))
+
+(setq cscope-excluded-packages '())
+
+(defun cscope/post-init-evil-jump ()
+  (defadvice helm-cscope-find-this-symbol (before cscope/goto activate)
+    (evil-jumper--push)))
+
+(defun cscope/init-xcscope ()
+  (use-package xcscope
+    :config
+    ;; for python projects, we don't want xcscope to rebuild the databse,
+    ;; because it uses cscope instead of pycscope
+    (setq-default cscope-option-do-not-update-database t)
+    (setq-default cscope-display-cscope-buffer nil)
+
+    (defun cscope//safe-project-root ()
+      "Return project's root, or nil if not in a project."
+      (and (fboundp 'projectile-project-root)
+           (projectile-project-p)
+           (projectile-project-root)))
+
+    (defun cscope/run-pycscope (directory)
+      (interactive (list (file-name-as-directory
+                          (read-directory-name "Run pycscope in directory: "
+                                               (cscope//safe-project-root)))))
+      (let ((default-directory directory))
+        (shell-command (format "pycscope -R -f '%s'"
+                               (expand-file-name "cscope.out" directory)))))
+
+    (evil-leader/set-key-for-mode 'python-mode "mgi" 'cscope/run-pycscope)
+    (evil-leader/set-key-for-mode 'c-mode "mgi" 'cscope-index-files)
+    (evil-leader/set-key-for-mode 'c++-mode "mgi" 'cscope-index-files)))
+
+(defun cscope/init-helm-cscope ()
+  (use-package helm-cscope
+    :config
+    (dolist (mode '(python-mode c-mode c++-mode))
+      (evil-leader/set-key-for-mode mode
+        "mgc" 'helm-cscope-find-called-function
+        "mgC" 'helm-cscope-find-calling-this-funtcion
+        "mgd" 'helm-cscope-find-global-definition
```

Comment:
Originally I did try to use the same bindings as gtags, but some just didn't fit well with cscope. Here are my reasons.

`helm-cscope-find-global-definition` on `mgd` instead of `mgg`:
- In gtags, `mgd` is `helm-gtags-find-tag` and `mgg` is `helm-gtags-dwim`. I think that `helm-cscope-find-global-definition` is more like `helm-gtags-find-tag` than `helm-gtags-dwim`, so I bound it to `mgd` and not `mgg`.
- AFAIK, helm-cscope doesn't have a dwim command such as `helm-gtags-dwim`.
- I don't want to override python's regular `mgg` which is `anaconda-mode-goto` and works very well.
- If we do want to bind `mgg`, I think `cscope-find-global-definition-no-prompting` (which I didn't bind) fits better than `helm-cscope-find-global-definition`.

`helm-cscope-find-called-function` and `helm-cscope-find-calling-this-function` on `mgc` and `mgC` instead of `mgi` and `mgI`:
- These are command that I think gtags doesn't have. The best equivalent for them is `helm-gtags-tags-in-this-function` which is bound to `mgi`. I did bind them originally to `mgi` and `mgI`, but I felt it wasn't mnemonic and intuitive as `mgc` and `mgC`.

`mgi` to create index database instead of `mgc`:
- Moved from `mgc` in favor of the find-called/calling commands. It's still kind of mnemonic - "I" for "index" instead of "C" for "create".

`helm-cscope-find-this-text-string` on `mgx` instead of `mgt`:
- `mgt` is reserved for "go to test" or something like that
- `x` is used for "text" in other places as well

So these were my reasons for the bindings. If you still think the bindings need to change, tell me and I'll change them.